public/js/navigator-js/integration/StateViewMap.js

```diff
@@ -43,33 +45,90 @@ this.navigatorjs.integration = this.navigatorjs.integration || {};
 				state = recipeStates[j];
 
 				if (requestedState.contains(state)) {
-					viewInstance = recipe.getViewInstance();
 
-					if (viewInstance.navigatorBehaviors instanceof Array) {
-						_addViewElementToDOM(recipe);
+					_addViewElementToDOM(recipe);
+					var viewInstance = recipe.getViewInstance();
+					if (Array.isArray(viewInstance.navigatorBehaviors)) {
 						_navigator.add(viewInstance, state);
 					}
+
+				} else {
+					if (recipe.isMounted()) {
+						_removeViewElementFromDOM(recipe);
+					}
 				}
 			}
 		}
 	}
 
+	function _removeViewElementFromDOM(recipe) {
```

Comment:
We are currently not removing the views that were created, but we are hiding them until they are transitionIn again. 
Based on the performances benchmark: 'Creation' is 5 to 30 times slower than 'Update'
If we decide to throw away our existing views and create them over and over, it will slow down significantly the page loading time.